### DISCUSSION TITLE
Basic Issue

### DISCUSSION THREAD

**Discussion by C4E5G3** [nix-community/kickstart-nix.nvim#39] [Q&A]
I can modify and run it as a flake locally, as per your instructions.

But when I modify the remote repository, and rebuild my system, my system does not pull in the updates.

**Comment by mrcjkb** [comment]
Hey 👋 

I don't understand the problem. Could you please be more specific?

**Reply by C4E5G3** [reply]
Hi!

Sorry CJ, I realize the problem...

but can I leave the discussion open for other questions?

**Reply by mrcjkb** [reply]
Sure.

**Reply by C4E5G3** [reply]
The configuration is fantastic CJ. The lualine is very smart, but I like 
a really obvious mode signal, so I installed lightline instead of lualine.
It is not as smart, but it is obvious.

I haven't put any parameters in lightline, but it works...

**Comment by C4E5G3** [comment]
I am trying to add Codeium, but it is unfree..I am not sure how instruct the build here.. 

```
Kickstart on 🐢 main [!] ❄️ 🌴🌴 nix run .
warning: Git tree '/home/marcus/Neovim/Kickstart' is dirty
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'neovim-0.10.0'
         whose name attribute is located at /nix/store/afawjm6nv8m5aagkcqxlxgrr5qpyh6cm-source/pkgs/stdenv/generic/make-derivation.nix:334:7

       … while evaluating attribute 'generatedWrapperArgs' of derivation 'neovim-0.10.0'

         at /nix/store/afawjm6nv8m5aagkcqxlxgrr5qpyh6cm-source/pkgs/applications/editors/neovim/wrapper.nix:106:14:

          105|       withRuby = rubyEnv != null;
          106|       inherit wrapperArgs generatedWrapperArgs;
             |              ^
          107|       luaRcContent = rcContent;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package ‘codeium-1.8.80’ in /nix/store/afawjm6nv8m5aagkcqxlxgrr5qpyh6cm-source/pkgs/by-name/co/codeium/package.nix:51 has an unfree license (‘unfree’), refusing to evaluate.

       a) To temporarily allow unfree packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNFREE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnfree = true; }
       in configuration.nix to override this.

       Alternatively you can configure a predicate to allow specific packages:
         { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
             "codeium"
           ];
         }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnfree = true; }
       to ~/.config/nixpkgs/config.nix.
```

**Reply by mrcjkb** [reply]
https://wiki.nixos.org/wiki/Unfree_software

**Reply by C4E5G3** [reply]
Yes, I tried this and I got some distorted text popping up. I tried again and for now seems better, but I am not sure that Codeium is actually installed. (I think I need to clear the cache from Nixvim)

So a flake input is just a module like everything else, and external data (e.g. in the configuration file)  can complete the flake?

**Comment by C4E5G3** [comment]
What I initially tried, and which you have confirmed, is to allow unfree software in my nixos configuration. But now I need to duplicate my flake to test it prior to deployment because, not being intrinsic to my configuration, it will always flag the unfree software error for the codeium plugin. So I need a copy of the flake minus codeium for testing.

**Reply by mrcjkb** [reply]
Is codeium a dependency of one of your Neovim plugins? 
If not, I would suggest to exclude it from your nvim flake and install it wherever you need it.
